### PR TITLE
DOC-3393: Document storage parameters

### PR DESCRIPTION
**Pull request by nickvigilante**
Fixes DOC-3393

### CODE REVIEW

**Review by ajwerner — APPROVED** [review]
Let's see if @otan wants to opine.

> Reviewed 3 of 6 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/14399#-:-N6J5iOoBKe9QHAf4LQH:bdnm4z5)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner and @nickvigilante)

___
*[`_includes/v22.1/misc/table-storage-parameters.md` line 7 at r1](https://reviewable.io/reviews/cockroachdb/docs/14399#-N6J5nH81hYXFAl7FpQ5:-N6J5nH9CFWWc_gv_5vP:b-k2eyav) ([raw file](https://github.com/cockroachdb/docs/blob/a9cbd501e083081cefe0755a17cfc4cb3cef6276/_includes/v22.1/misc/table-storage-parameters.md#L7)):*
> ```Markdown
> | ttl_automatic_column | If set, use the value of the `crdb_internal_expiration` hidden column. Currently always set to `true` and cannot be reset. | Boolean | `true` |
> | ttl_delete_batch_size | How many rows to [delete](delete.html) at a time. Minimum: 1. | Integer | 100 |
> | ttl_delete_rate_limit | Maximum number of rows to be deleted per second (rate limit). No theoretical limit. | Integer | 0 |
> ```

maybe note that `0` means no limit?

**Review by nickvigilante — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/14399)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner)

___
*[`_includes/v22.1/misc/table-storage-parameters.md` line 7 at r1](https://reviewable.io/reviews/cockroachdb/docs/14399#-N6J5nH81hYXFAl7FpQ5:-N6JOXdZ-fJqA3YIhM3l:bb6un1j) ([raw file](https://github.com/cockroachdb/docs/blob/a9cbd501e083081cefe0755a17cfc4cb3cef6276/_includes/v22.1/misc/table-storage-parameters.md#L7)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

maybe note that `0` means no limit?
</blockquote></details>

Done

**Review by otan — COMMENTED** [review]
> Reviewed 5 of 6 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/14399)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner, @nickvigilante, and @otan)

___
*[`_includes/v22.1/misc/index-storage-parameters.md` line 17 at r2](https://reviewable.io/reviews/cockroachdb/docs/14399#-N6KgeDX0e3sQwVLYcXf:-N6KgeDX0e3sQwVLYcXg:bxaq04d) ([raw file](https://github.com/cockroachdb/docs/blob/fedfd11f522efd5f7c9605bd1455a7e66e82d14e/_includes/v22.1/misc/index-storage-parameters.md#L17)):*
> ```Markdown
> | Parameter name |
> |---------------------|
> | autosummarize |
> ```

all of these except `fillfactor` (which doesn't seem included) return an unimplemented error.
___
*[`_includes/v22.1/misc/table-storage-parameters.md` line 16 at r2](https://reviewable.io/reviews/cockroachdb/docs/14399#-N6KgEWi2dqUzkXvGR-_:-N6KgEWi2dqUzkXvGR-a:bke13qs) ([raw file](https://github.com/cockroachdb/docs/blob/fedfd11f522efd5f7c9605bd1455a7e66e82d14e/_includes/v22.1/misc/table-storage-parameters.md#L16)):*
> ```Markdown
> | ttl_select_batch_size | How many rows to [select](select-clause.html) at one time during the row expiration check. Minimum: 1. | Integer | 500 |
> 
> The following parameters are included for PostgreSQL compatibility and do not affect how CockroachDB runs:
> ```

none of these will actually work except for `fillfactor` and `autovacuum_enabled`. they will return an unimplemented error.

**Review by otan — APPROVED** [review]

File: _includes/v22.1/misc/index-storage-parameters.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+| Parameter name | Description | Data type | Default value
+|---------------------+----------------------|-----|------|
+| bucket_count | Specifies the number of buckets into which a [hash-sharded index](hash-sharded-indexes.html) will split. | Integer | The value of the `sql.defaults.default_hash_sharded_index_bucket_count` [cluster setting](cluster-settings.html) |
```
Comment (stbof) [inline _includes/v22.1/misc/index-storage-parameters.md:3]:
```suggestion
| bucket_count | The number of buckets into which a [hash-sharded index](hash-sharded-indexes.html) will split. | Integer | The value of the `sql.defaults.default_hash_sharded_index_bucket_count` [cluster setting](cluster-settings.html) |
```

File: _includes/v22.1/misc/index-storage-parameters.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+| geometry_max_x | Maximum X-value of the [spatial reference system](spatial-glossary.html#spatial-reference-system) for the object(s) being covered. This only needs to be set if you are using a custom [SRID](spatial-glossary.html#srid). | | Derived from SRID bounds, else `(1 << 31) -1` |
```
Comment (stbof) [inline _includes/v22.1/misc/index-storage-parameters.md:4]:
```suggestion
| geometry_max_x | Maximum X-value of the [spatial reference system](spatial-glossary.html#spatial-reference-system) for the object(s) being covered. You need to set this only if you are using a custom [SRID](spatial-glossary.html#srid). | | Derived from SRID bounds, else `(1 << 31) -1` |
```

File: _includes/v22.1/misc/index-storage-parameters.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+| geometry_max_y | Maximum Y-value of the [spatial reference system](spatial-glossary.html#spatial-reference-system) for the object(s) being covered. This only needs to be set if you are using a custom [SRID](spatial-glossary.html#srid). | | Derived from SRID bounds, else `(1 << 31) -1` |
```
Comment (stbof) [inline _includes/v22.1/misc/index-storage-parameters.md:5]:
```suggestion
| geometry_max_y | Maximum Y-value of the [spatial reference system](spatial-glossary.html#spatial-reference-system) for the object(s) being covered. You need to set this only if you are using a custom [SRID](spatial-glossary.html#srid). | | Derived from SRID bounds, else `(1 << 31) -1` |
```

File: _includes/v22.1/misc/index-storage-parameters.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+| geometry_min_x | Minimum X-value of the [spatial reference system](spatial-glossary.html#spatial-reference-system) for the object(s) being covered. This only needs to be set if the default bounds of the SRID are too large/small for the given data, or SRID = 0 and you wish to use a smaller range (unfortunately this is currently not exposed, but is viewable on <https://epsg.io/3857>). By default, SRID = 0 assumes `[-min int32, max int32]` ranges. | | Derived from SRID bounds, else `-(1 << 31)` |
```
Comment (stbof) [inline _includes/v22.1/misc/index-storage-parameters.md:6]:
```suggestion
| geometry_min_x | Minimum X-value of the [spatial reference system](spatial-glossary.html#spatial-reference-system) for the object(s) being covered. You need to set this only if the default bounds of the SRID are too large or small for the given data, or SRID = 0 and you wish to use a smaller range (unfortunately this is currently not exposed, but is viewable on <https://epsg.io/3857>). By default, SRID = 0 assumes `[-min int32, max int32]` ranges. | | Derived from SRID bounds, else `-(1 << 31)` |
```

File: _includes/v22.1/misc/index-storage-parameters.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+| geometry_min_y | Minimum Y-value of the [spatial reference system](spatial-glossary.html#spatial-reference-system) for the object(s) being covered. This only needs to be set if you are using a custom [SRID](spatial-glossary.html#srid). | | Derived from SRID bounds, else `-(1 << 31)` |
```
Comment (stbof) [inline _includes/v22.1/misc/index-storage-parameters.md:7]:
```suggestion
| geometry_min_y | Minimum Y-value of the [spatial reference system](spatial-glossary.html#spatial-reference-system) for the object(s) being covered. You need to set this only if you are using a custom [SRID](spatial-glossary.html#srid). | | Derived from SRID bounds, else `-(1 << 31)` |
```

File: _includes/v22.1/misc/table-storage-parameters.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
+| Parameter name      | Description | Data type | Default value |
+|---------------------+----------------------+-----+------|
+| exclude_data_from_backup | Excludes the data in this table from any future backups. | Boolean | `false` |
```
Comment (stbof) [inline _includes/v22.1/misc/table-storage-parameters.md:3]:
```suggestion
| exclude_data_from_backup | Exclude the data in this table from any future backups. | Boolean | `false` |
```

File: _includes/v22.1/misc/table-storage-parameters.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+| ttl | Signifies if a TTL is active. Automatically set and controls the reset of all TTL-related storage parameters. | N/A | N/A |
+| ttl_automatic_column | If set, use the value of the `crdb_internal_expiration` hidden column. Currently always set to `true` and cannot be reset. | Boolean | `true` |
+| ttl_delete_batch_size | How many rows to [delete](delete.html) at a time. Minimum: `1`. | Integer | `100` |
```
Comment (stbof) [inline _includes/v22.1/misc/table-storage-parameters.md:6]:
```suggestion
| ttl_delete_batch_size | The number of rows to [delete](delete.html) at a time. Minimum: `1`. | Integer | `100` |
```

File: _includes/v22.1/misc/table-storage-parameters.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+| ttl_delete_rate_limit | Maximum number of rows to be deleted per second (rate limit). `0` means no limit.. | Integer | `0` |
+| ttl_expire_after | The [interval](interval.html) when a TTL will expire. This parameter is required to enable TTL. Minimum value: `'1 microsecond'`.<br/><br/>Use `RESET (ttl)` to remove from the table. | Interval | N/A |
+| ttl_job_cron | Frequency at which the TTL job runs. | [CRON syntax](https://cron.help) | `'@hourly'` |
+| ttl_label_metrics | Controls whether or not [TTL metrics](row-level-ttl.html#ttl-metrics) are labelled by table name (at the risk of added cardinality). | Boolean | `false` |
```
Comment (stbof) [inline _includes/v22.1/misc/table-storage-parameters.md:10]:
```suggestion
| ttl_label_metrics | Whether or not [TTL metrics](row-level-ttl.html#ttl-metrics) are labelled by table name (at the risk of added cardinality). | Boolean | `false` |
```

File: _includes/v22.1/misc/table-storage-parameters.md [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+| ttl_pause | If set, stops the TTL job from executing. | Boolean | `false` |
+| ttl_range_concurrency | The Row-Level TTL queries split up scans by ranges, and this determines how many concurrent ranges are processed at a time. Minimum: 1. | Integer | 1 |
```
Comment (stbof) [inline _includes/v22.1/misc/table-storage-parameters.md:12]:
```suggestion
| ttl_range_concurrency |  The number of concurrent ranges processed at a time when Row-Level TTL queries are split up scans by ranges. Minimum: 1. | Integer | 1 |
```

File: _includes/v22.1/misc/table-storage-parameters.md [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+| ttl_row_stats_poll_interval | If set, counts rows and expired rows on the table to report as Prometheus metrics while the TTL job is running. Unset by default, meaning no stats are fetched and reported. | Interval | N/A |
+| ttl_select_batch_size | How many rows to [select](select-clause.html) at one time during the row expiration check. Minimum: 1. | Integer | 500 |
```
Comment (stbof) [inline _includes/v22.1/misc/table-storage-parameters.md:14]:
```suggestion
| ttl_select_batch_size | The number of rows to [select](select-clause.html) at one time during the row expiration check. Minimum: 1. | Integer | 500 |
```

File: v22.1/reset-storage-parameter.md [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+
+The `RESET (storage parameter)` [statement](sql-statements.html) reverts the value of a storage parameter on a table or an index to its default value.
+
+{{site.data.alerts.callout_info}}
+The `SET (storage parameter)` is a subcommand of [`ALTER TABLE`](alter-table.html) and [`ALTER INDEX`](alter-index.html).
+{{site.data.alerts.end}}
+
+## Syntax
+
+{% comment %} railroad diagrams here {% endcomment %}
```
Comment (stbof) [inline v22.1/reset-storage-parameter.md:16]: Missing remote include

File: v22.1/reset-storage-parameter.md [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+-------------+--------------------------------------------------------------------------
+  ttl_test   | CREATE TABLE public.ttl_test (
+             |     id UUID NOT NULL DEFAULT gen_random_uuid(),
+             |     description STRING NULL,
+             |     inserted_at TIMESTAMP NULL DEFAULT current_timestamp():::TIMESTAMP,
+             |     CONSTRAINT ttl_test_pkey PRIMARY KEY (id ASC)
+             | )
+(1 row)
+~~~
+
```
Comment (stbof) [inline v22.1/reset-storage-parameter.md:87]: Could remove a few empty lines here.

**Review by stbof — COMMENTED** [review]
See suggestions and comments.

File: v22.1/reset-storage-parameter.md (hunk 11 above)
Comment (nickvigilante) [reply v22.1/reset-storage-parameter.md:16]: Yep, I'm going to include those once https://github.com/cockroachdb/cockroach/pull/83529 gets merged.

File: _includes/v22.1/misc/table-storage-parameters.md (hunk 8 above)
Comment (stbof) [inline _includes/v22.1/misc/table-storage-parameters.md:10]:
```suggestion
| ttl_label_metrics | Whether or not [TTL metrics](row-level-ttl.html#ttl-metrics) are labelled by table name (at the risk of added cardinality). | Boolean | `false` |
```

File: _includes/v22.1/misc/table-storage-parameters.md (hunk 10 above)
Comment (stbof) [inline _includes/v22.1/misc/table-storage-parameters.md:14]:
```suggestion
| ttl_select_batch_size | The number of rows to [select](select-clause.html) at one time during the row expiration check. Minimum: 1. | Integer | 500 |
```

File: v22.1/with-storage-parameter.md [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+{% remote_include https://raw.githubusercontent.com/cockroachdb/generated-diagrams/release-22.1/grammar_svg/create_table_with_storage_param.html %}
+</div>
+
+## Command parameters
+
+| Parameter           | Description                                                                                                                |
+|---------------------+----------------------|
+| `table`             | The table to which you are setting the parameter.                                                                                         |
+| `index`             | The index to which you are setting the parameter.                                                                                         |
+| `parameter_name`    | The name of the storage parameter you are changing. See [Storage parameters](#list-of-storage-parameters) for a list of available parameters. |
```
Comment (stbof) [inline v22.1/with-storage-parameter.md:34]:
```suggestion
| `parameter_name`    | The name of the storage parameter. See [Storage parameters](#list-of-storage-parameters) for a list of available parameters. |
```

File: v22.1/set-storage-parameter.md [hunk 14]
```diff
@@ -1,28 +1,43 @@
...
+
+{% comment %} need alter index diagram here {% endcomment %}
+
+## Command parameters
+
+| Parameter           | Description                                                                                                                |
+|---------------------+----------------------|
+| `table`             | The table to which you are setting the parameter.                                                                                         |
+| `index`             | The index to which you are setting the parameter.                                                                                         |
+| `parameter_name`    | The name of the storage parameter you are changing. See [Storage parameters](#list-of-storage-parameters) for a list of available parameters. |
```
Comment (stbof) [inline v22.1/set-storage-parameter.md:30]:
```suggestion
| `parameter_name`    | The name of the storage parameter. See [Storage parameters](#list-of-storage-parameters) for a list of available parameters. |
```

File: v22.1/set-storage-parameter.md [hunk 15]
```diff
@@ -1,28 +1,43 @@
 ---
 title: SET (storage parameter)
-summary: SET (storage parameter) applies a storage parameter to a table.
+summary: SET (storage parameter) applies a storage parameter to a table or an index after table/index creation.
```
Comment (stbof) [inline v22.1/set-storage-parameter.md:3]:
```suggestion
summary: SET (storage parameter) applies a storage parameter to a table or an index after table or index creation.
```

**Review by stbof — CHANGES_REQUESTED** [review]
See suggestions.

**Pull request merged**